src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6842,21 +6855,53 @@ private BoundExpression ConvertToArrayIndex(BoundExpression index, SyntaxNode no
             return result;
         }
 
+        private BoundExpression TryImplicitConversionToArrayIndex(BoundExpression expr, WellKnownType wellKnownType, SyntaxNode node, DiagnosticBag diagnostics)
+        {
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            TypeSymbol type = GetWellKnownType(wellKnownType, ref useSiteDiagnostics);
```

Comment:
Given this isn't used can we use a discard here?

Reply:
Same here.